pkg/availability/multisigcollector/internal/parts/batchreconstruction/batchreconstruction.go

```diff
@@ -132,9 +134,17 @@ func IncludeBatchReconstruction(
 			return nil
 		}
 
-		if _, ok := state.RequestState[reqID]; !ok {
+		if req, ok := state.RequestState[reqID]; !ok {
 			// Ignore a message with an invalid or outdated request id.
 			return nil
+		} else if _, ok = req.Txs[batchID]; !ok {
+			// Ignore a message with a batchID that was not requested.
+			return nil
+		}
+
+		if txs == nil || sliceutil.Contains(txs, nil) {
```

Comment:
```suggestion
		if len(txs) == 0 || sliceutil.Contains(txs, nil) {
```
(works also for nil, and catches a non-nil empty list)

Reply:
You suggested to move the nil check to computeids.go, so not committing this

Reply:
Sorry I wasn't clear enough about that. What I meant by the `nil` check is checking if any contained transactions are `nil` (second part of this condition), not if the whole list is `nil`. As mentioned in the other comment, I'd suggest doing  only the first part here (`len(txs)` == 0), since it is the availability module that rejects empty lists, and not the mempool. The second part can be removed indeed, since the mempool implements it as well.